services/google_translate.py

```diff
@@ -0,0 +1,64 @@
+from dataclasses import dataclass
+from typing import Callable, List
+
+from googleapiclient.discovery import build, Resource
+from googleapiclient.http import HttpRequest
+from returns.io import impure_safe
+
+from config.settings import Settings
+
+
+def make_request(request: HttpRequest) -> dict:
+    result = request.execute()
+    return result
+
+
+def build_request(service: Resource, multiple_texts: List[str]) -> HttpRequest:
+    request: HttpRequest = service.translations().list(target='en', q=multiple_texts)
+    return request
+
+
+TRANSLATE_QUERY_DATA_BATCH_SIZE = 60
+
+
+@dataclass
+class BatchRequestsBuilder:
+    data: List[str]
+    service: Resource
+    batch_size: int = TRANSLATE_QUERY_DATA_BATCH_SIZE
+    build_request: Callable = build_request
+
+    def __iter__(self):
+        self.current_index = 0
+        return self
+
+    def __next__(self):
+        if self.data[self.current_index:]:
+            start_index = self.current_index
+            self.current_index += self.batch_size
+            end_index = self.current_index
+            return self.build_request(self.service, self.data[start_index:end_index])
+        else:
+            raise StopIteration
+
+
+@dataclass
+class GoogleTranslateReader:
+    # Injected dependencies
+    settings: Settings
+    make_request: Callable = make_request
+    build_request: Callable = build_request
+
+    @impure_safe
+    def translate(self, source_text: List[str]) -> List[str]:
+        service = build('translate', 'v2', developerKey=self.settings.developer_key)
+        requests = BatchRequestsBuilder(source_text, service)
+        response = [self.make_request(request) for request in requests]
```

Comment:
You make as many HTTP requests as there are strings in the `source_text` list?

Reply:
No.

BatchRequestsBuilder will chunk source_text to batches.
for example, if we've got 100 strings in source_text, 
and batch_size is 60(by default),
then we'll get 2 requests, one with 60 strings, and another with 40 strings.

Reply:
Join strings with newlines to translate in bulk don't work,
since 1. Google return translated text without newlines 2. multiple languages in that bulk makes auto-detecting impossible

The good news is that we can post a list of strings with the API, but it has limits.
thus I made a BatchRequestsBuilder.

Reply:
Nice!